File: brand.alert/python/brand_alert_api_v2.py
try:
    import http.client as http
except ImportError:
    import httplib as http

import json

api_key = 'Your brand alert api key'

payload = {
    'includeSearchTerms': [
        'whois'
    ],
    'excludeSearchTerms': [
        '.win'
    ],
    'mode': 'purchase',
    'sinceDate': '2018-07-14',
    'apiKey': api_key,
    'responseFormat': 'json'
}


def print_response(txt):
    response_json = json.loads(txt)
    print(json.dumps(response_json, indent=4, sort_keys=True))


headers = {
    'Content-Type': 'application/json',
    'Accept': 'application/json'
}

conn = http.HTTPSConnection('brand-alert.whoisxmlapi.com')

conn.request(
        'POST',
        '/api/v2',
        json.dumps(payload),
        headers)

response = conn.getresponse()
text = response.read().decode('utf8')

print_response(text)

File: brand.alert/ruby/brand_alert_api_v2.rb
require 'json'
require 'net/https'
require 'openssl'
require 'uri'
require 'yaml' # only needed to print the returned result in a very pretty way

url = 'https://brand-alert.whoisxmlapi.com/api/v2'

########################
# Fill in your details #
########################
key = 'Your brand alert api key'

params = {
  includeSearchTerms: [
    'whois'
  ],
  excludeSearchTerms: [
    '.win',
    '.tz',
    ".com"
  ],
  mode: 'purchase',
  apiKey: key,
  responseFormat: 'json',
  sinceDate: '2018-07-10'
}

uri = URI.parse(url)
http = Net::HTTP.new(uri.host, uri.port)

# Connect using ssl
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE
request = Net::HTTP::Post.new(uri.request_uri)

# Set headers
request.add_field('Content-Type', 'application/json')
request.add_field('Accept', 'application/json')
request.body = params.to_json

# Get response
response = http.request(request)

# Print pretty parsed json
puts JSON.parse(response.body).to_yaml
